frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/SelectionManager.java

```diff
@@ -46,185 +45,217 @@
   private static final DefaultLogger LOGGER = new DefaultLogger(SelectionManager.class);
 
   private final ApplicationMaster am;
-  private final LinkedHashMap<String, Node> candidateNodes = new LinkedHashMap<>();
+  private final LinkedHashMap<String, Node> allNodes = new LinkedHashMap<>();
```

Comment:
>allNodes [](start = 44, length = 8)

Why remove word candidate here, but not remove in other places?

Reply:
all candidate was removed from member variable. Only keep on local  variable. 

---
In reply to: [174694706](https://github.com/Microsoft/pai/pull/318#discussion_r174694706) [](ancestors = 174694706)